### DISCUSSION TITLE
Creating training data for NER using SPACY 3.0 docbin

### DISCUSSION THREAD

**Discussion by Phat-Loc** [explosion/spaCy#8025] [Help: Coding & Implementations]
How should I create training data for NER in SPACY 3.0? I thought I could just create a Span from a doc and set a label then add it to the ents. Finally I was going to save the doc into a docbin for training. However if I do the following 

`span = doc[1:3]`
`span.label = 'entity_label_test'`

`Traceback (most recent call last):
  File "/home/phat/anaconda3/envs/spacy-extractors/lib/python3.9/site-packages/IPython/core/interactiveshell.py", line 3441, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-33-4074ba4dbf73>", line 1, in <module>
    span.label = 'entity_label_test'
  File "spacy/tokens/span.pyx", line 702, in spacy.tokens.span.Span.label.__set__
TypeError: an integer is required`

It looks like `span = Span(doc, 1, 3, label='entity_label_test')` works. I guess I should take the label and put it int he vocab to get an integer back and then use that when I want to set the span.label field?

**Comment by polm** [comment]
The important thing is to update `doc.ents`, not just spans in isolation. 

You should have a look at the [tutorial NER projects](https://github.com/explosion/projects/tree/v3/tutorials), which include training data.

**Comment by adrianeboyd** [comment]
On modifying spans: it looks like some inconsistencies got introduced into the `Span` API  in v3.0. We'll have a look! (To be clear: this is separate from how to create the training data.)
